From bcb12333ab38da6da459ef005c51a4c662f86392 Mon Sep 17 00:00:00 2001
From: zhangqing <zhangqing@rock-chips.com>
Date: Tue, 1 Dec 2015 03:39:27 -0800
Subject: [PATCH] ARM64: dts: pmic: add pmic init parameter for dts

add syr82x and act8846 pmic init parameters for dts.
enable syr82x and act8846 config by defult.

Change-Id: I8d77dc963cb038617b2b1817b435b67026b96de3
Signed-off-by: zhangqing <zhangqing@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3368-tb.dts | 142 +++++++++++++++++++++
 arch/arm64/configs/rockchip_defconfig      |   2 +
 2 files changed, 144 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3368-tb.dts b/arch/arm64/boot/dts/rockchip/rk3368-tb.dts
index d2da35524451..b7fc90931ed9 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-tb.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3368-tb.dts
@@ -76,6 +76,148 @@
 
 &i2c0 {
 	status = "okay";
+	syr827: syr827@40 {
+		compatible = "silergy,syr82x";
+		reg = <0x40>;
+		status = "okay";
+		regulators {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			syr827_dc1: regulator@0 {
+				reg = <0>;
+				regulator-compatible = "syr82x_dcdc1";
+				regulator-name = "vdd_arm";
+				regulator-min-microvolt = <712500>;
+				regulator-max-microvolt = <1500000>;
+				regulator-always-on;
+				regulator-boot-on;
+				regulator-initial-state = <3>;
+				regulator-state-mem {
+					regulator-mode = <0x2>;
+					regulator-off-in-suspend;
+					regulator-suspend-microvolt = <900000>;
+				};
+			};
+		};
+	};
+
+	syr828: syr828@41 {
+		compatible = "silergy,syr82x";
+		reg = <0x41>;
+		status = "okay";
+		regulators {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			syr828_dc1: regulator@0 {
+				reg = <0>;
+				regulator-compatible = "syr82x_dcdc1";
+				regulator-name = "vdd_gpu";
+				regulator-min-microvolt = <712500>;
+				regulator-max-microvolt = <1500000>;
+				regulator-always-on;
+				regulator-boot-on;
+				regulator-initial-state = <3>;
+				regulator-state-mem {
+					regulator-mode = <0x2>;
+					regulator-on-in-suspend;
+					regulator-suspend-microvolt = <900000>;
+				};
+			};
+		};
+	};
+
+	act8846: act8846@5a {
+		compatible = "active-semi,act8846";
+		reg = <0x5a>;
+		status = "okay";
+		system-power-controller;
+
+		regulators {
+			act8846_reg1: REG1 {
+				regulator-name = "VCC_DDR";
+				regulator-min-microvolt = <1200000>;
+				regulator-max-microvolt = <1200000>;
+				regulator-always-on;
+			};
+
+			act8846_reg2: REG2 {
+				regulator-name = "VCC_IO";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			act8846_reg3: REG3 {
+				regulator-name = "VDD_LOG";
+				regulator-min-microvolt = <700000>;
+				regulator-max-microvolt = <1500000>;
+				regulator-always-on;
+			};
+
+			act8846_reg4: REG4 {
+				regulator-name = "VCC_20";
+				regulator-min-microvolt = <2000000>;
+				regulator-max-microvolt = <2000000>;
+				regulator-always-on;
+			};
+
+			act8846_reg5: REG5 {
+				regulator-name = "VCCIO_SD";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			act8846_reg6: REG6 {
+				regulator-name = "VDD10_LCD";
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <1000000>;
+				regulator-always-on;
+			};
+
+			act8846_reg7: REG7 {
+				regulator-name = "VCCA_CODEC";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			act8846_reg8: REG8 {
+				regulator-name = "VCCA_TP";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			act8846_reg9: REG9 {
+				regulator-name = "VCCIO_PMU";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			act8846_reg10: REG10 {
+				regulator-name = "VDD_10";
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <1000000>;
+				regulator-always-on;
+			};
+
+			act8846_reg11: REG11 {
+				regulator-name = "VCC_18";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-always-on;
+			};
+
+			act8846_reg12: REG12 {
+				regulator-name = "VCC18_LCD";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-always-on;
+			};
+		};
+	};
 };
 
 &pwm0 {
diff --git a/arch/arm64/configs/rockchip_defconfig b/arch/arm64/configs/rockchip_defconfig
index e2aff5f87061..226a4f825056 100644
--- a/arch/arm64/configs/rockchip_defconfig
+++ b/arch/arm64/configs/rockchip_defconfig
@@ -283,6 +283,8 @@ CONFIG_ROCKCHIP_THERMAL=y
 CONFIG_MFD_RK808=y
 CONFIG_REGULATOR=y
 CONFIG_REGULATOR_FIXED_VOLTAGE=y
+CONFIG_REGULATOR_ACT8865=y
+CONFIG_REGULATOR_SYR82X=y
 CONFIG_MEDIA_SUPPORT=y
 CONFIG_MEDIA_CAMERA_SUPPORT=y
 CONFIG_MEDIA_RC_SUPPORT=y
-- 
2.35.3

